classes/ObjectModel.php

```diff
@@ -887,6 +887,29 @@ public function deleteSelection($ids)
         return $result;
     }
 
+    /**
+     * Does a soft delete on current object, using the "deleted" field in DB
+     * If the model object has no "deleted" property or no "deleted" definition field it will throw an exception
+     *
+     * @return bool
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function softDelete()
+    {
+        $definitions = ObjectModel::getDefinition($this);
+
+        if (empty($definitions['fields']['deleted'])) {
+            throw new PrestaShopException('Field "deleted" is missing from definition in object model ' . get_class($this));
+        }
+        if (!array_key_exists('deleted', $this)) {
```

Comment:
```suggestion
        if (!array_key_exists('deleted', get_object_vars($this))) {
```

Reply:
What about this? Does it still work in 7.4?

```suggestion
        if (!property_exists($this, 'deleted')) {
```

Reply:
thanx ! done (yes it should work in 7.4)

Reply:
Nope nope nope, the `property_exists` is also checking for private and protected property :scream:

Reply:
Since it's related to ObjectModel, maybe people are using `private $deleted;` :sweat_smile:

Reply:
![image](https://user-images.githubusercontent.com/1462701/85043044-358a1100-b18c-11ea-9697-ff59e75c50d1.png)

Reply:
Ok, good to know indeed!